### PR TITLE
[BREAK][IMPROVE] New emails design

### PR DESCRIPTION
**Pull request by ggazzo**
ps: its not done yet (even the prints) ;)

- [x] welcome screen
- [x] confirm email
- [x] mail messages
- [x] forgot password 
- [x] invitation email

- [ ] fix I18n

### welcome screen
image

### confirm email
image

### forgot password 
image

### mail messages
image

### invitation email
image

### CODE REVIEW

File: packages/rocketchat-mailer/package.js [hunk 1]
```diff
@@ -8,32 +8,10 @@ Package.onUse(function(api) {
 	api.use([
 		'ecmascript',
 		'ddp-rate-limiter',
-		'kadira:flow-router',
-		'rocketchat:lib',
-		'rocketchat:authorization',
 	]);
+	Npm.depends({
```
Comment (engelgabriel) [inline packages/rocketchat-mailer/package.js]: @ggazzo we are trying to keep the NPM dependencies on the root package.json file

@rodrigok can you help here?

File: packages/rocketchat-i18n/i18n/en.i18n.json [hunk 2]
```diff
@@ -555,6 +553,7 @@
   "Compact": "Compact",
   "Condensed": "Condensed",
   "Computer": "Computer",
+
```
Comment (engelgabriel) [inline packages/rocketchat-i18n/i18n/en.i18n.json]: remove empty line

File: packages/rocketchat-i18n/i18n/en.i18n.json [hunk 3]
```diff
@@ -1220,6 +1219,7 @@
   "Force_SSL": "Force SSL",
   "Force_SSL_Description": "*Caution!* _Force SSL_ should never be used with reverse proxy. If you have a reverse proxy, you should do the redirect THERE. This option exists for deployments like Heroku, that does not allow the redirect configuration at the reverse proxy.",
   "Forgot_password": "Forgot your password",
+  "Forgot_password_ask": "Forgot your password?",
```
Comment (engelgabriel) [inline packages/rocketchat-i18n/i18n/en.i18n.json]: Please don't add a new key, use the one above.

File: packages/rocketchat-i18n/i18n/en.i18n.json [hunk 4]
```diff
@@ -1663,6 +1664,7 @@
   "Log_View_Limit": "Log View Limit",
   "Logged_out_of_other_clients_successfully": "Logged out of other clients successfully",
   "Login": "Login",
+  "Login_now": "Login now",
```
Comment (engelgabriel) [inline packages/rocketchat-i18n/i18n/en.i18n.json]: Please don't add a new key, use the one above.

File: packages/rocketchat-i18n/i18n/en.i18n.json [hunk 5]
```diff
@@ -2146,6 +2148,7 @@
   "Require_password_change": "Require password change",
   "Resend_verification_email": "Resend verification email",
   "Reset": "Reset",
+  "Reset_now": "Reset now",
```
Comment (engelgabriel) [inline packages/rocketchat-i18n/i18n/en.i18n.json]: Please don't add a new key, use the one above.

File: packages/rocketchat-lib/server/functions/saveUser.js [hunk 6]
```diff
@@ -87,9 +88,18 @@ function validateUserData(userId, userData) {
 	}
 }
 
+let body = '';
+Meteor.startup(() => {
+	setTimeout(() => {
+		RocketChat.settings.get('Accounts_UserAddedEmail', (key, value) => {
+			body = inlinecss(value);
+			console.log(body);
```
Comment (engelgabriel) [inline packages/rocketchat-lib/server/functions/saveUser.js]: is this for debug?

**Comment by mnlbox** [comment]
@ggazzo Is it OK in RTL mode?

File: packages/rocketchat-i18n/i18n/en.i18n.json (hunk 5 above)
Comment (ggazzo) [reply packages/rocketchat-i18n/i18n/en.i18n.json]: @thiagosanchz what do you think about change that texts(remove the `now`)? I really like the now words :/

Comment (thiagosanchz) [reply packages/rocketchat-i18n/i18n/en.i18n.json]: @engelgabriel @ggazzo Why we can't add new words? I think the word NOW emphasizes the action.

Comment (engelgabriel) [reply packages/rocketchat-i18n/i18n/en.i18n.json]: @thiagosanchz @ggazzo @brunosquadros @sampaiodiego let's not create new translation keys like this, that are important and don't add much to the user experience. Every new KEY will have to be translated into 50 languages and maintained. We already have far more keys that we need and we have a big mess because of it. We are trying to reduce the number of redundant translation keys.

File: packages/rocketchat-channel-settings-mail-messages/server/methods/mailMessages.js [hunk 7]
```diff
@@ -1,9 +1,11 @@
 import _ from 'underscore';
 import moment from 'moment';
+import { send as sendEmail, checkEmail } from 'meteor/rocketchat:mailer';
```
Comment (engelgabriel) [inline packages/rocketchat-channel-settings-mail-messages/server/methods/mailMessages.js]: Let's nos use "as sendEmail" and keep the names the same everywhere for clarity.

File: packages/rocketchat-i18n/i18n/en.i18n.json [hunk 8]
```diff
@@ -1219,7 +1215,7 @@
   "Force_Disable_OpLog_For_Cache_Description": "Will not use OpLog to sync cache even when it's available",
   "Force_SSL": "Force SSL",
   "Force_SSL_Description": "*Caution!* _Force SSL_ should never be used with reverse proxy. If you have a reverse proxy, you should do the redirect THERE. This option exists for deployments like Heroku, that does not allow the redirect configuration at the reverse proxy.",
-  "Forgot_password": "Forgot your password",
+  "Forgot_password": "Forgot your password ?",
```
Comment (tassoevan) [inline packages/rocketchat-i18n/i18n/en.i18n.json]: Remove extra whitespace before the interrogation mark.

File: packages/rocketchat-i18n/i18n/en.i18n.json [hunk 9]
```diff
@@ -1610,6 +1606,7 @@
   "Leave_room": "Leave room",
   "Leave_Room_Warning": "Are you sure you want to leave the room \"%s\"?",
   "Leave_the_current_channel": "Leave the current channel",
+  "lets_get_you_new_one": "Let's get you new one!",
```
Comment (tassoevan) [inline packages/rocketchat-i18n/i18n/en.i18n.json]: `lets_...` -> `Lets_...`

File: packages/rocketchat-lib/server/functions/notifications/email.js [hunk 10]
```diff
@@ -111,14 +102,12 @@ export function sendEmail({ message, user, subscription, room, emailAddress, has
 
 	const link = getMessageLink(room, subscription);
 
-	if (RocketChat.settings.get('Direct_Reply_Enable')) {
-		contentFooter = RocketChat.placeholders.replace(RocketChat.settings.get('Email_Footer_Direct_Reply') || '');
-	}
+	const directReply = RocketChat.settings.get('Direct_Reply_Enable') ? '' : ''; // TODO templatemail
```
Comment (tassoevan) [inline packages/rocketchat-lib/server/functions/notifications/email.js]: Why is there a `TODO` here?
Do you need a migration to `Email_Footer_Direct_Reply` lose their footer-related content?

File: packages/rocketchat-lib/server/functions/saveUser.js [hunk 11]
```diff
@@ -1,6 +1,15 @@
 /* globals Gravatar */
 import _ from 'underscore';
 import s from 'underscore.string';
+import * as Mailer from 'meteor/rocketchat:mailer';
+let body = '';
+Meteor.startup(() => {
```
Comment (tassoevan) [inline packages/rocketchat-lib/server/functions/saveUser.js]: Why is this needed? If it is, why is it declared here?

File: packages/rocketchat-lib/server/functions/saveUser.js [hunk 12]
```diff
@@ -125,22 +133,9 @@ RocketChat.saveUser = function(userId, userData) {
...
-			let html;
-
-			if (RocketChat.settings.get('Accounts_UserAddedEmail_Customized')) {
-				subject = RocketChat.settings.get('Accounts_UserAddedEmailSubject');
-				html = RocketChat.settings.get('Accounts_UserAddedEmail');
-			} else {
-				subject = TAPi18n.__('Accounts_UserAddedEmailSubject_Default', { lng: user.language || RocketChat.settings.get('language') || 'en' });
-				html = TAPi18n.__('Accounts_UserAddedEmail_Default', { lng: user.language || RocketChat.settings.get('language') || 'en' });
-			}
+			const subject = RocketChat.placeholders.replace(RocketChat.settings.get('Accounts_UserAddedEmailSubject'));
```
Comment (tassoevan) [inline packages/rocketchat-lib/server/functions/saveUser.js]: Why not `Mailer.replace()`?

File: packages/rocketchat-mailer/server/api.js [hunk 13]
```diff
@@ -0,0 +1,86 @@
...
+let body;
+let Settings = {
+	get: () => {},
+};
+
+export const replacekey = (str, key, value = '') => str.replace(new RegExp(`(\\[${ key }\\]|__${ key }__)`, 'igm'), value);
+
+export const translate = (str) => str.replace(/\{ ?([^\} ]+)(( ([^\}]+))+)? ?\}/gmi, (match, key) => TAPi18n.__(key));
+
+export const replace = function replace(str, data = {}) {
```
Comment (tassoevan) [inline packages/rocketchat-mailer/server/api.js]: What if we don't expose this function? I mean, instead of `send()` accepting an object with a param-replaced `html` property, we call it passing the template and the parameters to be replaced inside the module. This can avoid any accidental non-escaping.

**Comment by rodrigok** [comment]
@ggazzo did you see the conflicts?

File: packages/rocketchat-i18n/i18n/en.i18n.json [hunk 14]
```diff
@@ -1299,6 +1298,7 @@
   "Help_Center": "Help Center",
   "Helpers": "Helpers",
   "Hex_Color_Preview": "Hex Color Preview",
+  "Hi_username": "Hi __name__",
   "Hidden": "Hidden",
   "Hide_Avatars": "Hide Avatars",
```
Comment (vynmera) [inline packages/rocketchat-i18n/i18n/en.i18n.json]: rest → reset

File: packages/rocketchat-i18n/i18n/en.i18n.json [hunk 15]
```diff
@@ -1453,9 +1454,7 @@
   "invisible": "invisible",
   "Invisible": "Invisible",
   "Invitation": "Invitation",
-  "Invitation_HTML": "Invitation HTML",
-  "Invitation_HTML_Default": "<h1>You have been invited to <strong>[Site_Name]</strong></h1><p>Go to [Site_URL] and try the best open source chat solution available today!</p>",
-  "Invitation_HTML_Description": "You may use the following placeholders: <br/><ul><li>[email] for the recipient email.</li><li>[Site_Name] and [Site_URL] for the Application Name and URL respectively.</li></ul>",
+  "Invitation_Email_Description": "You may use the following placeholders: <br/><ul><li>[email] for the recipient email.</li><li>[Site_Name] and [Site_URL] for the Application Name and URL respectively.</li></ul>",
```
Comment (vynmera) [inline packages/rocketchat-i18n/i18n/en.i18n.json]: get you _a_ new one

File: packages/rocketchat-i18n/i18n/en.i18n.json [hunk 16]
```diff
@@ -1634,6 +1633,7 @@
   "Leave_room": "Leave room",
   "Leave_Room_Warning": "Are you sure you want to leave the room \"%s\"?",
   "Leave_the_current_channel": "Leave the current channel",
```
Comment (vynmera) [inline packages/rocketchat-i18n/i18n/en.i18n.json]: create → created

on _the_ button

prefer "email address" over "address"

File: packages/rocketchat-i18n/i18n/en.i18n.json [hunk 17]
```diff
@@ -1346,6 +1346,7 @@
   "If_you_are_sure_type_in_your_password": "If you are sure type in your password:",
   "If_you_are_sure_type_in_your_username": "If you are sure type in your username:",
   "If_you_dont_have_one_send_an_email_to_omni_rocketchat_to_get_yours": "If you don't have one send an email to [omni@rocket.chat](mailto:omni@rocket.chat) to get yours.",
+  "If_you_didnt_ask_for_reset_ignore_this_email" : "If you didn't ask for reset your password, don't worry, just ignore this email",
```
Comment (geekgonecrazy) [inline packages/rocketchat-i18n/i18n/en.i18n.json]: > If you didn't ask for your password reset, don't worry, just ignore this email

would be more proper placement of reset.

> If you didn't ask for your password reset, you can ignore this email.

But to me this would be better :top:

File: packages/rocketchat-i18n/i18n/en.i18n.json (hunk 15 above)
Comment (geekgonecrazy) [reply packages/rocketchat-i18n/i18n/en.i18n.json]: Did this maybe end up on the wrong line?

File: packages/rocketchat-i18n/i18n/en.i18n.json [hunk 18]
```diff
@@ -2780,13 +2780,14 @@
   "UTF8_Names_Validation": "UTF8 Names Validation",
   "UTF8_Names_Validation_Description": "RegExp that will be used to validate usernames and channel names",
   "Validate_email_address": "Validate Email Address",
+  "Verification_email_body": "You have succesfully created an account on [Site_Name]. Please, click on the button bellow to confirm your email address and finish your registration",
```
Comment (geekgonecrazy) [inline packages/rocketchat-i18n/i18n/en.i18n.json]: bellow -> below

File: packages/rocketchat-i18n/i18n/en.i18n.json (hunk 17 above)
Comment (sampaiodiego) [reply packages/rocketchat-i18n/i18n/en.i18n.json]: I still don't think we should say to ignore 😛 it _might_ indicate a tentative of account steal.

File: packages/rocketchat-i18n/i18n/en.i18n.json (hunk 18 above)
Comment (sampaiodiego) [reply packages/rocketchat-i18n/i18n/en.i18n.json]: > bellow -> below

🤦‍♂️  I was so confident it was `bellow`

Comment (geekgonecrazy) [reply packages/rocketchat-i18n/i18n/en.i18n.json]: 😁 to Bellow is to yell or roar but in a deep voice
